cmd/frontend/backend/go_importers.go

```diff
@@ -62,109 +56,73 @@ func CountGoImporters(ctx context.Context, db database.DB, repo api.RepoName) (c
 		}
 	}()
 
-	ctx, cancel := context.WithTimeout(ctx, 20*time.Second) // avoid tying up resources unduly
-	defer cancel()
-
-	// Find all (possible) Go packages in the repository.
-	goPackages, err := listGoPackagesInRepoImprecise(ctx, db, repo)
-	if err != nil {
-		return 0, err
-	}
-	const maxSubpackages = 50 // arbitrary limit to avoid overloading api.godoc.org
-	if len(goPackages) > maxSubpackages {
-		goPackages = goPackages[:maxSubpackages]
+	var q struct {
+		Query     string
+		Variables map[string]interface{}
 	}
 
-	// Count importers for each of the repository's Go packages.
-	//
-	// TODO: The count sums together the user counts of all of the repository's subpackages. This
-	// overcounts the number of users, because if another project uses multiple subpackages in this
-	// repository, it is counted multiple times. This limitation is now documented and will be
-	// addressed in the future. See https://github.com/sourcegraph/sourcegraph/issues/2663.
-	for _, pkg := range goPackages {
-		// Assumes the import path is the same as the repo name - not always true!
-		req, err := http.NewRequest("GET", "https://api.godoc.org/importers/"+pkg, nil)
-		if err != nil {
-			return 0, err
-		}
-
-		response, err := countGoImportersHTTPClient.Do(req.WithContext(ctx))
-		if err != nil {
-			return 0, err
-		}
-		defer response.Body.Close()
-
-		var result struct {
-			Results []struct {
-				Path string
-			}
-		}
-		bytes, err := io.ReadAll(response.Body)
-		if err != nil {
-			return 0, err
-		}
-		err = json.Unmarshal(bytes, &result)
-		if err != nil {
-			return 0, err
-		}
-		count += len(result.Results)
+	q.Query = countGoImportersGraphQLQuery
+	q.Variables = map[string]interface{}{
+		"query": fmt.Sprintf(`f:go\.mod ^\s+%s\S*\s+v.* count:all visibility:public timeout:20s`, repo),
```

Comment:
Do we know how frequently this method is used? Would it be worthwhile having some sort of cache here?
This could generate quite some load. Maybe we should add a switch so we can act quickly if necessary?

Reply:
Is the input sanitized? Otherwise we might want to do something like this
```suggestion
		"query": fmt.Sprintf(`f:go\.mod ^\s+%s\S*\s+v.* count:all visibility:public timeout:20s`, regexp.QuoteMeta(repo)),
```

Reply:
> Is the input sanitized? Otherwise we might want to do something like this

Good call!